### PR TITLE
Rework examples to use a common foundation

### PR DESCRIPTION
**Pull request by wkharold**
This PR simplifies fluxfw-gcp example deployments by factoring the necessary network, IAM, and storage components out of the individual examples into a common foundation.

The foundation is deployed using a top-level Makefile.
The examples are deployed via Terraform

Multiple examples can be deployed on a single instance of the foundation.

### CODE REVIEW

File: fluxfw-gcp/tf/examples/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+```bash
+cd foundation
+make
+```
+
+Once the foundation is deployed you can visit any of the example directories and follow the steps
+in the their README documentation to deploy the example.
+
+_Note_ the foundation only needs to be deployed once. Any of the examples, or a collection of them,
+can be deployed simultaneously on the foundation.
```
Comment (mmm) [inline fluxfw-gcp/tf/examples/README.md:23]: I recommend providing very literal instructions for someone who won't actually _read_ this, but who might be clicking through.  I.e., subsequent commands that either assume they're still in the `foundation` dir or renavigate from the toplevel.

Also, how does this behave in the cloudshell tutorial mode?  That's sort of the ultimate test of someone not reading but clicking through the steps sequentially.

File: fluxfw-gcp/tf/examples/foundation/Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+NETWORK_STATE = network/terraform.tfstate
+IAM_STATE     = iam/terraform.tfstate
+STORAGE_STATE = storage/terraform.tfstate
```
Comment (mmm) [inline fluxfw-gcp/tf/examples/foundation/Makefile:3]: is there a way to make this makefile useful to a user who wants to "adopt" (copy/paste) this?  remote state's gonna be one of the first things they'll do on thier own project.  No-op, just thinking out-loud.

File: fluxfw-gcp/tf/examples/foundation/Makefile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+project_id = $$(gcloud config get-value core/project)
+region     = $$(gcloud config get-value compute/region)
```
Comment (mmm) [inline fluxfw-gcp/tf/examples/foundation/Makefile:6]: failure mode if gcloud isn't installed?

File: fluxfw-gcp/tf/examples/foundation/Makefile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+iam: $(IAM_STATE)
+
+$(IAM_STATE): $(NETWORK_STATE)
+	@cd iam && $(MAKE) .default
+
+.PHONY: storage
+storage: $(STORAGE_STATE)
+
+$(STORAGE_STATE): $(NETWORK_STATE)
+	@cd storage && $(MAKE) .filestore storage_name=$(storage_name) share_name=$(share_name)
```
Comment (mmm) [inline fluxfw-gcp/tf/examples/foundation/Makefile:31]: the `@`s make them silent right?  Perhaps you wanna stick some messages in here to show them what it's doing?  Balance of noise -vs- hidden magic

File: fluxfw-gcp/tf/examples/foundation/iam/Makefile [hunk 5]
```diff
@@ -0,0 +1,20 @@
+DEFAULT_CONFIG=default/main.tf default/outputs.tf default/state.tf
+TERRAFORM_STATE=terraform.tfstate terraform.tfstate.backup
+TERRAFORM_FILES=default/.terraform default/.terraform.lock.hcl
+
+.PHONY: init
+init: 
+	terraform -chdir=default init
```
Comment (mmm) [inline fluxfw-gcp/tf/examples/foundation/iam/Makefile:7]: failure mode for no terraform?

**Comment by mmm** [comment]
this is a huge change... I'd generally recommend someone break this up into smaller, more single-purposed PRs.

No-op, just had time to give this a litte more serious review than usual.

**Review by mmm — APPROVED** [review]
lgtm

**Comment by wkharold** [comment]
Oops, I didn't merge this to main before I did the PR. I'll close this, merge, and do a new PR.

**Pull request closed without merging**